test/unit/StakingProducts/initializeProducts.js

```diff
@@ -86,44 +162,100 @@ describe('initializeProducts', function () {
     expect(await stakingProducts.getTotalEffectiveWeight(poolId)).to.be.equal(2000);
+  });
 
-    await stakingProducts.recalculateEffectiveWeightsForAllProducts(poolId);
-    const totalProducts = await cover.productsCount();
-    expect(totalProducts).to.be.gt(validProducts.length);
-    for (let i = 0; i < totalProducts; i++) {
-      const product = await stakingProducts.getProduct(poolId, i);
-      if (i < validProducts.length) {
-        expect(product.lastEffectiveWeight).to.be.equal(validProducts[i].weight);
-      } else {
-        expect(product.lastEffectiveWeight).to.be.equal(0);
-      }
-    }
-    {
-      const weights = await stakingProducts.weights(poolId);
-      expect(weights.totalTargetWeight).to.be.equal(2000);
-      expect(weights.totalEffectiveWeight).to.be.equal(2000);
-      expect(await stakingProducts.getTotalTargetWeight(poolId)).to.be.equal(2000);
-      expect(await stakingProducts.getTotalEffectiveWeight(poolId)).to.be.equal(2000);
+  it('should fail to increase target weight when effective weight is at the limit', async function () {
+    const { stakingProducts, stakingPool, cover } = this;
+    const [manager, staker, coverBuyer] = this.accounts.members;
+    const [internalContract] = this.accounts.internalContracts;
+
+    // Impersonate cover contract
+    const coverSigner = await ethers.getImpersonatedSigner(cover.address);
+    await setEtherBalance(cover.address, parseEther('100000'));
+
+    const coverId = 1;
+    const amount = parseEther('10000');
+
+    // Get capacity in staking pool
+    await depositTo.call(this, { staker, amount });
+
+    // 200 products with 5 weight = 50% of max weight (200 * 5 = 1000 / 2000)
+    const products = Array(200)
+      .fill('')
+      .map((value, index) => {
+        return { ...product0, productId: index, weight: 5 };
+      });
+
+    // Add products
+    await stakingProducts.connect(internalContract).setInitialProducts(poolId, products);
+
+    // Buy all available cover for every product
+    const allocationPromises = [];
+    for (let i = 0; i < products.length; i++) {
+      allocationPromises.push(
+        cover.allocateCapacity(
+          { ...buyCoverParamsTemplate, productId: i, owner: coverBuyer.address, amount: amount.div(10) },
+          coverId,
+          0,
+          stakingPool.address,
+        ),
+      );
     }
+    await Promise.all(allocationPromises);
+
+    // total target and total effective weight should be at the max
+    expect(await stakingProducts.getTotalTargetWeight(poolId)).to.be.equal(1000);
+    expect(await stakingProducts.getTotalEffectiveWeight(poolId)).to.be.equal(1000);
+
+    // Burn 75% of the current stake
+    // Effective weight was at 50%, so with 3/4 of capacity reduced, allocations are twice as much as capacity
+    // ie. 50/100 = 1000 effective weight, burn 75% of stake -> 50/25 = 4000 effective weight
+    const activeStake = await stakingPool.getActiveStake();
```

Comment:
not clear why this further logic is tested here in this test and file

so initializeProducts is a file that focuses on that function only

to me looks like this logic should be tested in 

recalculateEffectiveWeights.js